File: Day 21 - 30/Day-21/pendulum_array.py
# Program to Print the Array in to-and-fro motion

def pendulum_array(array, size):
    array.sort()
    simple_pendulum = [0] * size

    # Midlle Term
    mid_term = (size - 1) // 2
    i = 1
    j = 1

    # Sorting as Pendulum Motion
    simple_pendulum[mid_term] = array[0]
    for _ in range(1, mid_term + 1):

        simple_pendulum[mid_term + i] = array[j]
        j += 1

        simple_pendulum[mid_term - i] = array[j]
        j += 1

        i += 1

    # Correction if Size of the Array is Even
    if size % 2 == 0:
        simple_pendulum[mid_term + i] = array[j]

    return simple_pendulum

def main():
    import os
    os.system("cls")

    print("\n\n\t\t Simple Pendulum Motion !")
    size =int(input("\n\n Enter the Size of the Array: "))
    print("\n Enter Element One-by-One: \n")
    
    array = [int(input()) for _ in range(size)]
    result = pendulum_array(array, size)
    print("\n\n Simple Pendulum motion:\n")
    print(*result)

if __name__ == "__main__":
    main()

File: Day 21 - 30/Day-21/pendulum_array.c
#include <stdio.h>
#include <conio.h>
#include <stdlib.h>

void pendulum_array(int[], int);

int main(void)
{
    int size;
    system("cls");

    printf("\n\n\t\t Simple Pendulum Motion !");
    printf("\n\n Enter the Size of the Array: ");
    scanf("%d", &size);

    int array[size], result[size];
    printf("\n Enter Element One-by-one:\n\n");
    for(int i = 0; i < size; i++) scanf("%d", &array[i]);

    pendulum_array(array, size);
}

void pendulum_array(int array[], int size)
{
    int sp[size], temp = 0, i, j;

    // Sorting the Array
    for(i = 0; i < size; i++)
    {
        for(j = i + 1; j < size; j++)
        {
            if(array[i] > array[j])
            {
                temp = array[i];
                array[i] = array[j];
                array[j] = temp;
            }
        }
    }

    // Creating Simple Pendulum Array !
    for(int i = 0; i < size; i++) sp[i] = 0;

    // Middle Term
    int mid = (size - 1)/2;
    sp[mid]  = array[0];
    i = 1;
    j = 1;

    // Sorting as Pendulum Motion
    for(i = 1; i <= mid; i++)
    {
        sp[mid + i] = array[j++];
        sp[mid - i] = array[j++];
    }

    // Correction if Size of the Array is Even
    if(size%2 == 0) sp[mid + i] = array[j];

    printf("\n\n Pendulum Motion: ");
    for(i = 0; i < size; i++) printf("%d ", sp[i]);
    getch();
}